tests/components/tplink/test_light.py

```diff
@@ -0,0 +1,230 @@
+"""Tests for light platform."""
```

Comment:
Most of the tests are seemingly testing internal behavior of the pyhs100 library (e.g., the query_helper parts look very similar what pyhs100 does internally) and will break soon with 0.40 release. If tests are wanted for the platform, they should not depend on how the library behaves internally but should merely test the external behavior.

Reply:
Makes sense. However, mocking the external methods can hide incompatabilites between the component and the module. 
End to end tests in this manner will ensure the component will work when changes are made to the module. It may be redundant but it provides the closest real world test.

Reply:
The idea with mocking is to describe the expected behavior (and used functionality) instead of the internal workings of the used library, to my understanding at least. Therefore I'd prefer if you would not pollute homeassistant's codebase with checks similar that already exist in upstream.